samples/nrf9160/aws_fota/src/main.c

```diff
@@ -33,12 +33,13 @@ BUILD_ASSERT_MSG(!IS_ENABLED(CONFIG_LTE_AUTO_INIT_AND_CONNECT),
 #endif
 
 #if !defined(CONFIG_CLOUD_CLIENT_ID)
-#define IMEI_LEN 15
-#define CLIENT_ID_LEN (IMEI_LEN + 4)
+/* Define the length of the IMEI AT COMMAND response buffer */
+#define IMEI_LEN 19
+#define CLIENT_ID_LEN (IMEI_LEN + sizeof("nrf-"))
 #else
-#define CLIENT_ID_LEN (sizeof(CONFIG_CLOUD_CLIENT_ID) - 1)
+#define CLIENT_ID_LEN sizeof(CONFIG_CLOUD_CLIENT_ID)
 #endif
-static u8_t client_id_buf[CLIENT_ID_LEN+1];
+static u8_t client_id_buf[CLIENT_ID_LEN + 1];
```

Comment:
Curious: Why `+ 1`?

Reply:
It's for the addition of `\0`  termination in the end of the buffer so that you could use `CLIENT_ID_LEN` to get the length of the string but I should probably swap it to be `sizeof`.

Reply:
sizeof sounds like a good idea